cns/restserver/internalapi.go

```diff
@@ -151,6 +152,37 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host version in container status.
+// If NMAgent updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(channelMode string) {
+	for i, containerstatus := range service.state.ContainerStatus {
+		// Will open a separate PR to convert all the NC version related variable to int. Change from string to int is a pain.
+		hostVersion, err := strconv.Atoi(containerstatus.HostVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.HostVersion %s to int, err msg %v", containerstatus.HostVersion, err)
+			return
+		}
+		vmVersion, err := strconv.Atoi(containerstatus.VMVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.VMVersion %s to int, err msg %v", containerstatus.VMVersion, err)
+			return
+		}
+		// host NC version is the NC version from NMAgent, if it's already keep up with NC version exist in VM, no update needed.
+		if hostVersion >= vmVersion {
+			continue
+		} else {
+			newHostNCVersion := service.imdsClient.GetNetworkContainerInfoFromHostWithoutToken()
+			service.Lock()
+			if channelMode == cns.KubernetesCRD {
+				service.UpdatePendingProgrammingIPs(strconv.Itoa(newHostNCVersion), containerstatus.CreateNetworkContainerRequest)
+			}
+			containerstatus.HostVersion = strconv.Itoa(newHostNCVersion)
+			service.state.ContainerStatus[i] = containerstatus
```

Comment:
This line is not required. You are updating the object directly. So need to reassign it.

Reply:
This line is needed. In golang, when updating map value, reassign is needed. Please refer this link https://stackoverflow.com/questions/42716852/how-to-update-map-values-in-go
![image](https://user-images.githubusercontent.com/16043262/97904992-24ced780-1cf6-11eb-80a6-6b704ba51d25.png)